extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/EventBusInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,160 @@
+
+        return serverBuilder
+                .addAttributesExtractor(eventBusAttributesExtractor)
+                .newConsumerInstrumenter(new TextMapGetter<>() {
+                    @Override
+                    public Iterable<String> keys(final Message message) {
+                        return message.headers().names();
+                    }
+
+                    @Override
+                    public String get(final Message message, final String key) {
+                        if (message == null) {
+                            return null;
+                        }
+                        return message.headers().get(key);
+                    }
+                });
+    }
+
+    static Instrumenter<Message, Message> getProducerInstrumenter(final OpenTelemetry openTelemetry) {
+        EventBusAttributesExtractor eventBusAttributesExtractor = new EventBusAttributesExtractor(SEND);
+
+        InstrumenterBuilder<Message, Message> serverBuilder = Instrumenter.builder(
+                openTelemetry,
+                INSTRUMENTATION_NAME, MessagingSpanNameExtractor.create(eventBusAttributesExtractor));
+
+        return serverBuilder
+                .addAttributesExtractor(eventBusAttributesExtractor)
+                .newProducerInstrumenter((message, key, value) -> {
+                    if (message != null) {
+                        message.headers().set(key, value);
+                    }
+                });
+    }
+
+    private static class EventBusAttributesExtractor extends MessagingAttributesExtractor<Message, Message> {
+        private final MessageOperation operation;
+
+        public EventBusAttributesExtractor(final MessageOperation operation) {
+            this.operation = operation;
+        }
+
+        @Override
+        public MessageOperation operation() {
+            return operation;
+        }
+
+        @Override
+        protected String system(final Message message) {
+            return "vert.x";
+        }
+
+        @Override
+        protected String destinationKind(final Message message) {
+            return "topic";
+        }
+
+        @Override
+        protected String destination(final Message message) {
+            return message.address();
+        }
+
+        @Override
+        protected boolean temporaryDestination(final Message message) {
+            return false;
+        }
+
+        @Override
+        protected String protocol(final Message message) {
+            return null;
+        }
+
+        @Override
+        protected String protocolVersion(final Message message) {
+            return null;
+        }
+
+        @Override
+        protected String url(final Message message) {
+            return null;
+        }
+
+        @Override
+        protected String conversationId(final Message message) {
+            return null;
+        }
+
+        @Override
+        protected Long messagePayloadSize(final Message message) {
+            return null;
```

Comment:
That could be computed, however, it only makes sense in distributed mode.

Reply:
I think we can leave it empty for now. It is not a mandatory attribute per the OTel spec.